src/models/experts/asyncActions.ts

```diff
@@ -47,13 +47,69 @@ export const fetchExperts = createAsyncThunk(
       return {
         expertIds: appendIds,
         experts,
-        meta: {
-          pageNumber: response.data.number,
-          totalPages: response.data.totalPages,
-          totalElements: response.data.totalElements,
-          isLastPage: response.data.last,
-          appendExperts,
+        totalPages: response.data.totalPages,
+        totalElements: response.data.totalElements,
+        isLastPage: response.data.last,
+        appendExperts,
+      };
+    } catch (err) {
+      return rejectWithValue(err.response?.data);
+    }
+  },
+);
+
+export const fetchExpertsAutorsList = createAsyncThunk(
+  'experts/loadExpertsAuthorsList',
+  async (options: IExpertsAutorsListOptions, { getState, rejectWithValue }) => {
+    try {
+      const {
+        experts: {
+          data,
+          meta: {
+            size,
+            textFields: { author },
+            sort: { order, sortBy },
+          },
+        },
+      } = getState() as any;
+
+      const isSearchByAuthorexists = (arg: string) => {
+        if (!arg) {
+          return `${sortBy.split(',')}` + `,${order}`;
```

Comment:
when we sort by fullName, on the server side the is no such key. Only firstName/lastName exist. In this case we should combine two keys for one column fullName and while sending the request pass multiple values for one parameter. This could be soloved by passing  array of param values [val1, val2] - that is the reason of using split()

Reply:
so sortBy can be: 'firstName', 'lastName' or 'fullName' and for last one we should pass to the server 'firstName,lastName'. Is it correct? If yes, and we always append `,${order}` can we just ``return `${sortBy},${order}`;`` ? 

Ideally, I'd expect some mapping on UI side, like 

```
sortTypesMap = {
  firstName: ['firstName'],
  fullName: ['firstName', 'lastName']
}
```

so we could have search presets instead of hardcoding, but this is rather low-value-higher-efforts thing. So I'm not sure if it makes sense even to discuss by now.

Reply:
done